src/providers/whatsapp/whatsapp.service.ts

```diff
@@ -17,59 +19,66 @@ export class WhatsappService {
   constructor(private readonly socketGateway: SocketGateway) {
     const appConfig = getAsyncAppConfig();
 
-    // create({
-    //   session: appConfig.appname,
-    //   logQR: false,
-    //   catchQR: this.onWaitQrCode,
-    // })
-    //   .then((client) => {
-    //     this.client = client;
-    //     this.socketGateway.broadcast('init', 'Sessão Criada');
-    //     console.log('client');
-    //   })
-    //   .catch(() => {
-    //     console.log('Erro ao criar instancia do whatsapp');
-    //   });
+    create({
+      session: appConfig.appname,
+      logQR: true,
+      catchQR: this.onWaitQrCode,
+    })
+      .then((client) => {
+        this.client = client;
+
+        process.on('SIGINT', function () {
+          client.close();
+        });
+      })
+      .catch(() => {
+        console.log('Erro ao criar instancia do whatsapp');
+      });
   }
 
   onWaitQrCode: CatchQR = (qrCode) => {
-    this.socketGateway.broadcast(
-      EventTypes.QRCODE,
-      'Aguardando leitura do QRCode',
-    );
-    console.log('qrCode');
+    this.socketGateway.broadcast(EventTypes.QR_CODE, qrCode);
   };
 
+  async listenOnConnection() {
+    if (await this.client.isConnected()) {
+      this.onConnection();
+    }
+  }
+
+  onConnection() {
+    this.socketGateway.broadcast(
+      EventTypes.CONNECTION_STATUS,
+      DefaultMessages.CONNECT,
+    );
+  }
```

Comment:
Qual o objetivo desse onConnection?

- Informar o status do whatsapp?
- Informar o status do websocket?

Lembra que o onConnection é um método reservado do socketIo que executa toda vez que alguem é conectado no socket

Reply:
Informar o status do whatsapp? isso, eu posso mudar o nome

Reply:
Mas se for isso, não da pra ti usar esse método onConnection(), porque ele é disparado toda vez q o socketio muda de estado

Reply:
Tu tem q receber o status do whatsapp do venom e repassar via socket, assim como foi feito com o qrcode